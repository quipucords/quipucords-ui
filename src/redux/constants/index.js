import { aboutModalTypes } from './aboutModalConstants';
import { confirmationModalTypes } from './confirmationModalConstants';
import { credentialsTypes } from './credentialsConstants';
import { factsTypes } from './factsConstants';
import { reportsTypes } from './reportsConstants';
import { scansTypes } from './scansConstants';
import { sourcesTypes } from './sourcesConstants';
import { statusTypes } from './statusConstants';
import { toastNotificationTypes } from './toastNotificationConstants';
import { userTypes } from './userConstants';
import { viewTypes } from './viewConstants';

const reduxTypes = {
  aboutModal: aboutModalTypes,
  confirmationModal: confirmationModalTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  status: statusTypes,
  toastNotifications: toastNotificationTypes,
  user: userTypes,
  view: viewTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  aboutModalTypes,
  confirmationModalTypes,
  credentialsTypes,
  factsTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  statusTypes,
  toastNotificationTypes,
  userTypes,
  viewTypes
};
