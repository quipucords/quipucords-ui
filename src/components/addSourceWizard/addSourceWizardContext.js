import { useShallowCompareEffect } from 'react-use';
import { AlertVariant } from '@patternfly/react-core';
import { reduxTypes, storeHooks } from '../../redux';
import apiTypes from '../../constants/apiConstants';
import { translate } from '../i18n/i18n';

/**
 * Return an updated source. Display relative toast messaging after wizard closes.
 *
 * @param {object} options
 * @param {Function} options.t
 * @param {Function} options.useDispatch
 * @param {Function} options.useSelector
 * @returns {{}}
 */
const useGetAddSource = ({
  t = translate,
  useDispatch: useAliasDispatch = storeHooks.reactRedux.useDispatch,
  useSelector: useAliasSelector = storeHooks.reactRedux.useSelector
} = {}) => {
  const selectorResponse = useAliasSelector(({ addSourceWizard }) => addSourceWizard, {});
  const dispatch = useAliasDispatch();
  const { edit, error, errorMessage, fulfilled, show, source } = selectorResponse;

  useShallowCompareEffect(() => {
    if (show === false && (fulfilled || error)) {
      dispatch({
        type: reduxTypes.toastNotifications.TOAST_ADD,
        alertType: (error && AlertVariant.danger) || AlertVariant.success,
        header: t('toast-notifications.title_add-source_hidden', {
          context: [error && 'error', edit && 'edit']
        }),
        message: t('toast-notifications.description_add-source_hidden', {
          context: [error && 'error', edit && 'edit'],
          message: errorMessage,
          name: source?.[apiTypes.API_SUBMIT_SOURCE_NAME]
        })
      });
    }
  }, [dispatch, edit, error, fulfilled, show, source, t]);

  return selectorResponse;
};

const context = {
  useGetAddSource
};

export { context as default, context, useGetAddSource };
