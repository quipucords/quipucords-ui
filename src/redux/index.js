import React from 'react';
import { connect } from 'react-redux';
import { useLocation, useNavigate, useParams } from 'react-router-dom';
import { store } from './store';
import { reduxActions } from './actions';
import { reduxHelpers } from './common';
import { storeHooks } from './hooks';
import { reduxReducers } from './reducers';
import { reduxSelectors } from './selectors';
import { reduxTypes } from './constants';

/**
 * Patch withRouter for legacy components.
 *
 * @param {React.ReactNode} Component
 * @returns {React.ReactNode}
 */
const withRouter = Component => props => {
  const location = useLocation();
  const navigate = useNavigate();
  const params = useParams();
  return (
    <Component
      {...props}
      history={{ push: navigate }}
      location={location}
      navigate={navigate}
      params={params}
      router={{ location, navigate, params }}
    />
  );
};

const connectRouter = (mapStateToProps, mapDispatchToProps) => component =>
  withRouter(connect(mapStateToProps, mapDispatchToProps)(component));

export {
  connect,
  connectRouter,
  reduxActions,
  reduxHelpers,
  reduxReducers,
  reduxSelectors,
  reduxTypes,
  store,
  storeHooks
};
