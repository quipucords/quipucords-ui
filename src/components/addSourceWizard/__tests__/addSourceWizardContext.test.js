import { context, useGetAddSource } from '../addSourceWizardContext';
import apiTypes from '../../../constants/apiConstants';
import { store } from '../../../redux';

describe('AddSourceWizardContext', () => {
  let mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.spyOn(store, 'dispatch').mockImplementation((type, data) => ({ type, data }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return specific properties', () => {
    expect(context).toMatchSnapshot('specific properties');
  });

  it('should apply a hook for retrieving data from a selector', async () => {
    const { result: errorResponse } = await mountHook(() =>
      useGetAddSource({
        useSelector: () => ({
          error: true,
          show: false,
          data: {
            messages: {}
          }
        })
      })
    );

    expect(errorResponse).toMatchSnapshot('error response');

    const { result: errorShowResponse } = await mountHook(() =>
      useGetAddSource({
        useSelector: () => ({
          error: true,
          data: {
            messages: {}
          }
        })
      })
    );

    expect(errorShowResponse).toMatchSnapshot('error show response');

    const { result: successResponse } = await mountHook(() =>
      useGetAddSource({
        useSelector: () => ({
          fulfilled: true
        })
      })
    );

    expect(successResponse).toMatchSnapshot('success response');

    const { result: successShowResponse } = await mountHook(() =>
      useGetAddSource({
        useSelector: () => ({
          fulfilled: true,
          show: false,
          source: {
            [apiTypes.API_SUBMIT_SOURCE_NAME]: 'lorem ipsum'
          }
        })
      })
    );

    expect(successShowResponse).toMatchSnapshot('success show response');
    expect(mockDispatch.mock.calls).toMatchSnapshot('dispatch confirmations on hide wizard');
  });
});
