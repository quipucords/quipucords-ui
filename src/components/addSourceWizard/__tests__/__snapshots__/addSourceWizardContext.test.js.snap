// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSourceWizardContext should apply a hook for retrieving data from a selector: dispatch confirmations on hide wizard 1`] = `
[
  [
    {
      "alertType": "danger",
      "header": "t(toast-notifications.title_add-source_hidden, {"context":"error"})",
      "message": "t(toast-notifications.description_add-source_hidden, {"context":"error"})",
      "type": "TOAST_ADD",
    },
  ],
  [
    {
      "alertType": "success",
      "header": "t(toast-notifications.title_add-source_hidden, {"context":""})",
      "message": "t(toast-notifications.description_add-source_hidden, {"context":"","name":"lorem ipsum"})",
      "type": "TOAST_ADD",
    },
  ],
]
`;

exports[`AddSourceWizardContext should apply a hook for retrieving data from a selector: error response 1`] = `
{
  "data": {
    "messages": {},
  },
  "error": true,
  "show": false,
}
`;

exports[`AddSourceWizardContext should apply a hook for retrieving data from a selector: error show response 1`] = `
{
  "data": {
    "messages": {},
  },
  "error": true,
}
`;

exports[`AddSourceWizardContext should apply a hook for retrieving data from a selector: success response 1`] = `
{
  "fulfilled": true,
}
`;

exports[`AddSourceWizardContext should apply a hook for retrieving data from a selector: success show response 1`] = `
{
  "fulfilled": true,
  "show": false,
  "source": {
    "name": "lorem ipsum",
  },
}
`;

exports[`AddSourceWizardContext should handle an onShowAddSourceWizard event: onColumnSort event, dispatch 1`] = `
[
  [
    {
      "type": "CREATE_SOURCE_SHOW",
    },
  ],
]
`;

exports[`AddSourceWizardContext should return specific properties: specific properties 1`] = `
{
  "useGetAddSource": [Function],
  "useOnShowAddSourceWizard": [Function],
}
`;
